mlir/lib/Dialect/MIOpen/Pipeline/Pipeline.cpp

```diff
@@ -46,6 +46,10 @@ void miopen::addHighLevelPipeline(PassManager &pm, bool toMIOpen) {
     pm.addPass(tosa::createTosaToMIOpenPass());
   }
   mlir::tosa::addTosaToLinalgPasses(pm);
+  bufferization::OneShotBufferizationOptions options;
+  options.allowReturnAllocs = true;
+  options.testAnalysisOnly = true;
+  pm.addPass(createLinalgComprehensiveModuleBufferizePass(options));
   pm.addPass(createLinalgElementwiseOpFusionPass());
   pm.addPass(createLinalgBufferizePass());
```

Comment:
Many of these will not become unnecessary. Only bufferization::createBufferResultsToOutParamsPass is still needed for Bufferization.